### DISCUSSION TITLE
Effect presets easy to accidentally delete.

### DISCUSSION THREAD

**Discussion by gudjonbe** [wwmm/easyeffects#2277] [General]
Good day developers!

I have been using this app for over a year now, and I quite like it! Good Job!

However, there are two things that bother me about the effect-preset curtain. 
1) The "DELETE" button is to close to the "SAVE" button, and too often have I accidentally clicked it. 
2) When the preset-curtain is open, I cannot focus on any other window, until I close the curtain. It is quite annoying.

Have a great day!
Gudjon

**Comment by wwmm** [comment]
>The "DELETE" button is to close to the "SAVE" button, and too often have I accidentally clicked it.

Which EasyEffects version are you using? In the last release the distance between buttons was increased 
![Screenshot from 2023-04-17 11-31-42](https://user-images.githubusercontent.com/16225172/232517959-1e1a4936-531f-4bd8-b5bb-db8426d050a1.png)

>When the preset-curtain is open, I cannot focus on any other window, until I close the curtain. It is quite annoying.

That is a gtk4 bug affecting mostly Xorg users #1182.

**Reply by gudjonbe** [reply]
Wow, that was a quick response! 

1) Yes, this looks exactly like the version I'm using (7.0.3). I don't like how proud the delete button is. I would rather want to right-click it and get some options, like 'rename', 'delete', 'lock'. It is also a bit annoying, how easy it is to accidentally save your current effect to the one you want to load. 
    Every time I edit my effects, I have to slow down, and make sure I'm indeed pressing the right button.
Maybe a different tab for managing the effects, or a prompt? (No, everyone hates prompts)... I don't know.

BTW, is there an option to create keyboard shortcuts for changing effects? (similar to FX-sound on W10)

2) I think the focus-bug is no longer an issue. Seems to be OK now.

**Reply by wwmm** [reply]
>Maybe a different tab for managing the effects, or a prompt? (No, everyone hates prompts)... I don't know.

The is a discussion about those buttons on #1678. As this menu is already a little crowded I am not sure another tab is going to be a solution because the input effects pipeline is also in the menu. And the prompt also has the gtk4 technical difficulties I talked about on #1678. So a good solution is not going to be easy to get.

>BTW, is there an option to create keyboard shortcuts for changing effects? (similar to FX-sound on W10)

There isn't because the complexity involved in this is not small. Nowadays we can put in the pipeline multiple instances of a given plugin. Managing these shortcuts in a consistent way is probably going to be a nightmare.

**Reply by gudjonbe** [reply]
Hey, I don't know exactly when this was updated, but I REALLY like the new prompt before I accidentally delete my presets. Thanks!

**Reply by wwmm** [reply]
Thanks! :-) I do not remember when but it has been implemented for a few months. I had the idea after seeing how ChatGPT's site handled the history removal confirmation. I totally forgot to close this discussion.
